### PR TITLE
(DOCSP-14577): UUID

### PR DESCRIPTION
**Pull request by mongomoe**
### Jira

- https://jira.mongodb.org/browse/DOCSP-14577

### Staged Changes (Requires MongoDB Corp SSO)

- [NODE.JS SDK > Data Types > UUID](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-14577-uuid/sdk/node/data-types/uuid/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by dacharyc — APPROVED** [review]
LGTM! Nice example showing usage.

File: source/sdk/node/data-types/embedded-objects.txt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+Overview
+--------
+
+An embedded object is a special type of :ref:`Realm object <node-object-schemas>`
+that models complex data about a specific object. Embedded objects are similar
+to :ref:`relationships <node-client-relationships>`, but they provide additional
+constraints and map more naturally to the denormalized :manual:`MongoDB document
+model </core/data-modeling-introduction/>`.
+
+Realm enforces unique ownership constraints that treat each embedded object as
```
Comment (kneth) [inline source/sdk/node/data-types/embedded-objects.txt:24]: If you want to be fancy, you can say it is a [bijective mapping](https://en.wikipedia.org/wiki/Bijection). Or you can say that it is an object with a property which is an object (`{ a: { b: "c" } }`).

File: source/sdk/node/data-types/collections.txt [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+{+service-short+} has several types to represent groups of objects,
+which we call **collections**. A collection is an object that contains
+zero or more instances of one :ref:`{+service-short+} type
+<node-object-types>`. {+service-short+} collections are **homogenous**:
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/node/data-types/collections.txt]: Fair point, I've removed the sentence "{+service-short+} collections are **homogenous**:
all objects in a collection are of the same type." now.

Also, note: this PR was supposed to be targeting 'new-data-types-node-rn', but was accidentally targeting 'master', which is why you see outdated changes to files unrelated to UUID. I've altered the target to 'new-data-types-node-rn' now.

File: source/sdk/node/data-types/collections.txt [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+Like :ref:`live objects <node-live-objects>`, {+service-short+} collections
+are usually **live**:
+
+- Live results collections always reflect the current results of the associated query.
+- Live lists always reflect the current state of the relationship on the {+realm+} instance.
+
+A collection is **not** live when: 
+
+- it is a :ref:`results collection <node-realm-results>` that you are iterating through using a :mdn:`for..in <Web/JavaScript/Reference/Statements/for...in>` or :mdn:`for..of <Web/JavaScript/Reference/Statements/for...of>` statement. Both statements will continue to iterate through objects in the collection even if you have deleted or modified the collection's objects to exclude them from the filter that produced the results collection.
+- the collection is a frozen :js-sdk:`Results.snapshot() <Realm.Collection.html#snapshot>`.
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/node/data-types/collections.txt]: Would the following alternate phrasing be okay?:
```
"A collection is **not** live when: 

- ... {same first bullet as before}
-  the collection's length is preserved through a snapshot() but it's elements may be mutated."
```

File: source/sdk/node/data-types/embedded-objects.txt (hunk 1 above)
Comment (mongomoe) [reply source/sdk/node/data-types/embedded-objects.txt]: I would prefer to use examples in the docs that have more of a narrative, and can tell users what kind of use-cases they might use a specific feature for. For instance, the `{ a: { b: "c" } }` doesn't tell a user what kind of data would be nested (other than the fact that 'b' is an object that is inside an object, a. However, the [example](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-14577-uuid/sdk/node/data-types/embedded-objects/#realm-object-models) below gives the user a specific narrative that they might be able to relate to or have a similar use case in their applications. 

In addition, simply saying that embedded objects are objects that are a property of an object might give the readers the wrong impression that they do not have constraints, such as schema validation, when working with embedded objects.

Comment (kneth) [reply source/sdk/node/data-types/embedded-objects.txt]: Yes, let's keep it short and clear. Using too much math will not make it easier for people to read.

**Review by kneth — APPROVED** [review]

File: source/sdk/node/data-types/embedded-objects.txt (hunk 1 above)
Comment (mongomoe) [reply source/sdk/node/data-types/embedded-objects.txt]: 👍

**Pull request merged**